### DISCUSSION TITLE
message_type / dns_entries

### DISCUSSION THREAD

**Discussion by uggyuggy** [d-Rickyy-b/certstream-server-go#31] [Q&A] [answered]
Thank you very much for providing an alternative to Calidog !

New user, I'm starting testing using the  `domains_only_url` (as interrested for now only by domains ;) )
I see into the json output, every `data` with domain names seems to come also with a `"message_type": "dns_entries"`.

I read one of your comments into an issue:
> If you only need domains, connect to the /domains-only endpoint. That saves a lot of bandwidth and parsing.

Which makes sense.

So I wonder if I have missed something, if that is something wanted, or if that is possible to have this/an endpoint not sending that additional  `"message_type": "dns_entries"` for each certificate, to save some bandwidth and parsing ?

Thank's !

**Comment by d-Rickyy-b** [comment, marked as answer]
Hi there, thanks for your suggestion.

> if that is something wanted, or if that is possible to have this/an endpoint not sending that additional  "message_type": "dns_entries" for each certificate, to save some bandwidth and parsing ?

In theory absolutely possible. I created certstream-server-go as a drop-in replacement for Calidog's server, so I took everything as it was there and simply copied it into this tool. That's being done for backwards compatibility.

See also https://github.com/CaliDog/certstream-server/blob/60de7000901e5eb246d2e83c908678b43e5a60c8/lib/certstream/client_manager_agent.ex#L83-L86

I think I could add some configuration to disable the "message_type" there. Although compared to full certificates the amount of data transferred is already quite reduced already.

**Comment by uggyuggy** [comment]
Hi Rico,

Ohhh  ok. That makes sense now. Thank's for the explanation and link.
I tested your server only with another "raw" websocket client.

> I think I could add some configuration to disable the "message_type" there. 

Could be great, thx :+1:
